lib/rubocop/cop/root_cops/factory_file_name.rb

```diff
@@ -28,6 +70,27 @@ def on_send(node)
         def _factory_method?(method)
           method == :factory
         end
+
+        def _invariant_noun?(word)
+          word.pluralize == word.singularize
+        end
+
+        def _whitelisted?(word)
+          WHITELIST.include?(word)
+        end
+
+        def _file_name_has_error?
+          offenses.first&.severity&.name == :fatal
+        end
```

Comment:
I decided that if the file name is incorrectly named then we don't want to inspect the factory names within that file because their validity rely on comparison with a valid file name. Because of this, a [factory file name error is fatal](https://github.com/Root-App/root-ruby-style/blob/2d322fbc18e4e02297e1cd9ca0055b538f7a89c0/lib/rubocop/cop/root_cops/factory_file_name.rb#L29) rather than being a standard error. So, if we see that within the file's `offenses` array there is a fatal error then we will skip the rest of the file (it seems that every file being inspected by rubocop has an `offenses` array that is initially empty`).